### ISSUE TITLE
dplyr 1.0.0 dev version breaks get_decennial when length(variables) > 48

### ISSUE THREAD

**Issue by jlegewie** [walkerke/tidycensus#229]
Hi,

I installed the dev version of dplyr 1.0.0, which is [coming out soon](https://www.tidyverse.org/blog/2020/03/dplyr-1-0-0-is-coming-soon/). I think tidycensus will have to be updated. The problem I ran into happens in `get_decennial` with `length(variables) > 48`. `bind_cols` in that if condition now renames some of the variables to avoid duplicate columns for `NAME` and `GEOID`, which produces an error down the line.

One way to deal with this problem is to change the code in the if condition to this but I am sure there are more elegant solutions.

```
l <- split(variables, ceiling(seq_along(variables)/48))
        dat <- map(l, function(x) {
            d <- try(load_data_decennial(geography, x, key, year, 
                sumfile, state, county, show_call = show_call), 
                silent = TRUE)
            if (inherits(d, "try-error")) {
                d <- try(suppressMessages(load_data_decennial(geography, 
                  x, key, year, sumfile = "sf3", state, county, 
                  show_call = show_call)))
            }
            d
        })
        dat[-1] <- dat[-1] %>% map(select, -NAME, -GEOID)
        dat <- bind_cols(dat)
```

**Comment by walkerke** [comment]
Thank you for the note!  I'm digging into this and I'll have a fix ready before dplyr 1.0 comes out.

**Comment by mfherman** [comment]
I've been looking at this a bit too -- one simple fix is replacing `bind_cols` with a `reduce(left_join)` so that the `NAME` and `GEOID` aren't duplicated. This works when geometry is not requested, but was failing for me with sf geometry due to [this issue](https://github.com/tidyverse/dplyr/issues/4917). But, this seems to have been resolved in the most recent version of dplyr. I'm currently fighting with reinstalling sf and other dependencies from source on my machine, so it might be working now! Will report back when I get everything up and running again.

``` r
  if (length(variables) > 48) {
    l <- split(variables, ceiling(seq_along(variables) / 48))

    dat <- map(l, function(x) {
      d <- try(load_data_decennial(geography, x, key, year, sumfile, state, county, show_call = show_call),
                 silent = TRUE)
      # If sf1 fails, try to get it from sf3
      if (inherits(d, "try-error")) {
        d <- try(suppressMessages(load_data_decennial(geography, x, key, year, sumfile = "sf3", state, county, show_call = show_call)))
      }
      d
    }) %>%
      reduce(left_join, by = c("GEOID", "NAME"))
```

**Comment by walkerke** [comment]
Addressed in PR #233
